clang/lib/CConv/DeclRewriter.cpp

```diff
@@ -197,7 +222,8 @@ void DeclRewriter::rewriteFieldOrVarDecl(DRType *N, RSet &ToRewrite) {
 }
 
 void DeclRewriter::rewriteSingleDecl(DeclReplacement *N, RSet &ToRewrite) {
-  assert("Declaration is not a single declaration." && isSingleDeclaration(N));
+  bool isSingleDecl = dyn_cast<TypedefDecl>(N->getDecl()) || isSingleDeclaration(N);
+  assert("Declaration is not a single declaration." && isSingleDecl);
```

Comment:
What is this check doing? Why are typedefs treated specially?

Reply:
Typedefs don't have one of the fields that `isSingleDecleration` checks for.

Reply:
I see. So if it's a typedef you just assume that it's equivalent to being a single declaration, whatever that means?